clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1026,7 +986,7 @@ public Map<String, List<PartitionInfo>> listTopics(Duration timeout) {
             final AllTopicsMetadataEvent topicMetadataEvent = new AllTopicsMetadataEvent(timer);
```

Comment:
Also here the timer is only used in the event. Using a deadline would be simpler.

Reply:
Changed.